src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -86,21 +86,24 @@ public void discoverClusterSlots(Jedis jedis) {
       rediscovering = true;
 
       try {
-        this.slots.clear();
-
         List<Object> slots = jedis.clusterSlots();
 
+        // We should clear slots after getting result about cluster slots, because if we got exception or timeout in
```

Comment:
Maybe we want to hold a lock during trying all nodes. 
I see what you said, we release the write lock before completing slot cache update so there's race again.